dpcomp_gpu_runtime/lib/gpu_runtime.cpp

```diff
@@ -346,7 +346,9 @@ struct Stream {
       throw std::runtime_error("Failed to allocate MemInfo");
     }
 
-    retain();
+    #if defined(IMEX_ENABLE_NUMBA_HOTFIX)
```

Comment:
@nbpatel why do we even have numba_dpcomp-specific things in `gpu_runtime`? I am fine with the quick fix, but we need a plan to refactor/redesign the code.

Reply:
I guess it is there for the python frontend. Agree, we need to refactor it